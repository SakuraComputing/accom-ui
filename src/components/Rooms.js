import React from 'react';
import { connect } from 'react-redux';
import Room from './Room';
import facilityList from '../common/facilityList';

export class Rooms extends React.Component {

    constructor(props) {
        super(props);
        this.getRooms = this.getRooms.bind(this);
    }

    getRooms() {
        const [ { rooms } ] = this.props.accommodations.filter(accommodation => accommodation.id === parseInt(this.props.match.params.id));
        let RoomItem;
        return RoomItem = rooms.map(room => (<Room key={room.id} {...room}/>))
    }

    render() {

        return (
            <div className="content-container">
                <h2>Rooms</h2>
                {this.getRooms()}
                { this.props.facilities ? facilityList(this.props.facilites) : ''}
            </div>
        )
    }
}
const mapStateToProps = (state) =>  ({
   accommodations: state.accommodations
});
export default connect(mapStateToProps, null)(Rooms);