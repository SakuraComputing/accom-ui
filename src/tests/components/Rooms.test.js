import React from 'react';
import { shallow } from 'enzyme';
import Rooms from '../../components/Rooms';

describe('Rooms Page', () => {
   it('should render the form correctly', () => {
      const wrapper = shallow(<Rooms/>);
      expect(wrapper).toMatchSnapshot()
   });
});
