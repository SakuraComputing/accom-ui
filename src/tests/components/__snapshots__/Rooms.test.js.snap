// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rooms Page should render the form correctly 1`] = `
<div>
  <h1>
    Rooms
  </h1>
</div>
`;
